impeller/display_list/paint.h

```diff
@@ -36,12 +33,7 @@ struct Paint {
 
   /// @brief Whether or not a save layer with the provided paint can perform the
   ///        opacity peephole optimization.
-  static bool CanApplyOpacityPeephole(const Paint& paint) {
-    return paint.blend_mode == BlendMode::kSourceOver &&
-           paint.invert_colors == false &&
-           !paint.mask_blur_descriptor.has_value() &&
-           paint.image_filter == nullptr && paint.color_filter == nullptr;
-  }
+  static bool CanApplyOpacityPeephole(const Paint& paint);
```

Comment:
SaveLayer operations ignore the color source. If this is having an impact then something must be going wrong in how we handle SaveLayers.